packages/api/src/sdk-bundle-api/sdk-to-bundle/session-data.ts

```diff
@@ -101,3 +108,33 @@ export interface EarlyRequest {
   startTime: number;
   duration: number;
 }
+
+export interface IDBObjectStoreMetadata {
+  databaseName: string;
+  objectStoreName: string;
+  serialize?: (value: any) => string;
+  deserialize?: (value: string) => any;
+}
+
+/**
+ * Currently we only support string keys, but we may support other types in the future.
+ * Keys are:
+ * - Serialized in recorder/src/storage/storage.ts
+ * - Deserialized in puppeteer-utils/src/browser-context/storage.ts
+ *
+ * @see IDBValidKey for all possible types.
+ */
+export type SerializedIDBValidKey = StringKey;
+
+interface StringKey {
+  type: "string";
+  serializedKey: string;
+}
+
+export type IDBObjectStoreWithEntries = Omit<
+  IDBObjectStoreMetadata,
+  "serialize" | "deserialize"
+> & {
+  createObjectStoreOptions: IDBObjectStoreParameters;
+  entries: { key?: SerializedIDBValidKey; value: string }[];
```

Comment:
```suggestion
  /**
   * Entry values are JSON stringified. JSON representable objects, arrays and primitives are supported,
   * all other types (e.g. Date, Regex, ArrayBuffer etc.) are not yet supported.
   */
  entries: { key?: SerializedIDBValidKey; value: string }[];
```

Reply:
Also mentioned that a customer [de]serialize function can be defined to handle unsupported types.